client/blocks/reader-author-link/docs/example.jsx

```diff
@@ -9,16 +9,17 @@ import React from 'react';
 import ReaderAuthorLink from 'blocks/reader-author-link';
 import Card from 'components/card';
 
-export default React.createClass( {
+const ReaderAuthorLinkExample = () => {
 
-	displayName: 'ReaderAuthorLink',
+	const author = { URL: 'http://wpcalypso.wordpress.com', name: 'Barnaby Blogwit' };
 
-	render() {
-		const author = { URL: 'http://wpcalypso.wordpress.com' };
-		return (
-			<Card>
-				<ReaderAuthorLink author={ author }>Author site</ReaderAuthorLink>
-			</Card>
-		);
-	}
-} );
+	return (
+		<Card>
+			<ReaderAuthorLink author={ author }>{ author.name }</ReaderAuthorLink>
```

Comment:
Thanks for the update :)
From my view, I love how you make it consistent with other link components, and also make it a pure component to be consistent with our current recommend style.